drivers/adc/adc_stm32.c

```diff
@@ -342,21 +346,40 @@ static int adc_stm32_dma_start(const struct device *dev,
 		LOG_ERR("Problem setting up DMA: %d", ret);
 		return ret;
 	}
+	
+	uint32_t dma_transfer_mode = 0;
+	if (ctx->options.batch_mode) {
+		/* The ADC will be set in continuous mode so that it will
+		 * continue sampling until the DMA transfered enough samples,
+		 * which will trigger DMA TC interrupt, which we will use to
+		 * stop the sampling.
+		 */
+		dma_transfer_mode = LL_ADC_REG_DMA_TRANSFER_UNLIMITED;
+		LL_ADC_REG_SetContinuousMode(adc, LL_ADC_REG_CONV_CONTINUOUS);
+	} else {
+		/* ADC set to one-shot mode because all channels will
+		 * only read once by this DMA request
+		 */
+		dma_transfer_mode = LL_ADC_REG_DMA_TRANSFER_LIMITED;
```

Comment:
Actually it should be fine to use both LIMITED and UNLIMITED transfer mode here, as long as you don't enable continuous mode. Not all STM32 series support LIMITED mode, and you do stop the transfer in dma_callback anyway.

Reply:
You are right, I was also thinking about that. Currently my UNLIMITED mode doesn't fully work yet so I'm still debugging what's going on. But will likely change it like you're suggesting to keep the driver simple.

Reply:
I think we could actually decouple transfer mode from batch mode and always use LIMITED for the "new" series (on these chips UNLIMITED is needed only for circular transfers which you don't use) and UNLIMITED for the "old" ones in single mode.